app/models/hbx_enrollment.rb

```diff
@@ -1780,10 +1780,57 @@ def can_terminate_coverage?
+    return true if benefit_sponsorship.benefit_applications.approved_and_term_benefit_applications_by_date(reinstate_date).present?
+
+    ::EnrollRegistry.feature_enabled?(:admin_shop_end_date_changes) ? benefit_sponsorship&.prior_py_benefit_application&.benefit_sponsor_catalog&.effective_period&.cover?(reinstate_date) : false
+  end
+
+  def has_active_term_or_expired_exists_for_reinstated_date?
+    reinstate_date = fetch_reinstatement_date
+    is_shop? ? shop_active_term_or_expired_for_reinstated_date(reinstate_date) : ivl_active_term_or_expired_for_reinstated_date(reinstate_date)
+  end
+
+  def shop_active_term_or_expired_for_reinstated_date(reinstate_date)
+    eligible_states = ENROLLED_AND_RENEWAL_STATUSES + CAN_REINSTATE_AND_UPDATE_END_DATE
+    eligible_states << 'coverage_expired' if ::EnrollRegistry.feature_enabled?(:admin_shop_end_date_changes)
+    application = benefit_sponsorship.benefit_applications.approved_and_term_benefit_applications_by_date(reinstate_date).first
+    return false unless application
+
+    application_effective_period = application.effective_period
+    HbxEnrollment.by_effective_period(application_effective_period).where({:family_id => self.family_id,
+                                                                           :kind.in => %w[employer_sponsored employer_sponsored_cobra],
+                                                                           :effective_on.gte => reinstate_date,
+                                                                           :coverage_kind => self.coverage_kind,
+                                                                           :employee_role_id => self.employee_role_id,
+                                                                           :aasm_state.in => eligible_states}).any?
+  end
+
+  def ivl_active_term_or_expired_for_reinstated_date(reinstate_date)
+    eligible_states = ENROLLED_AND_RENEWAL_STATUSES + CAN_REINSTATE_AND_UPDATE_END_DATE
+    eligible_states << 'coverage_expired' if ::EnrollRegistry.feature_enabled?(:admin_ivl_end_date_changes)
+    benefit_coverage_period = HbxProfile.current_hbx.benefit_sponsorship.benefit_coverage_period_by_effective_date(reinstate_date)
+    effective_period = benefit_coverage_period.start_on..benefit_coverage_period.end_on
+    HbxEnrollment.by_effective_period(effective_period).where({:family_id => self.family_id,
+                                                               :kind.in => %w[individual coverall],
+                                                               :effective_on.gte => reinstate_date,
+                                                               :coverage_kind => self.coverage_kind,
+                                                               :consumer_role_id => consumer_role_id,
+                                                               :aasm_state.in => eligible_states}).any?
   end
 
   def has_active_or_term_exists_for_reinstated_date?
```

Comment:
we can remove this method if we no longer using it

Reply:
I am using this method